src/bootstrap/lib.rs

```diff
@@ -233,6 +234,7 @@ impl Build {
         };
         let rust_info = channel::GitInfo::new(&src);
         let cargo_info = channel::GitInfo::new(&src.join("cargo"));
+        let src_is_git = src.join(".git").is_dir();
```

Comment:
Hmm, I thought it was better to be more precise, but I see your point.  I'll check just for existence.